tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -315,6 +315,13 @@ internal static void SetTagObject(Span span, string key, object? value, bool all
                 return;
             }
 
+            // remap keys as needed regardless of datatype
+            if (key == "http.response.status_code")
+            {
+                span.SetTag(Tags.HttpStatusCode, value.ToString());
```

Comment:
Do we want to add additional type checking here to ensure that `value` is some numerical value?

Reply:
Do we really want this? 🤔 I strongly suspect this would break the expectations the backend has about the format of the status code? 🤔 What happens when we set the status code to `200.0`? I'm 80% sure it's Not Good™, let alone if it's `OK`

Reply:
> Do we really want this? 

In my view the current `int` only approach (or if it is an older version and only has `string` tags) is correct as that is _what_ the HTTP Status Code is.

But `system-tests` convert the JSON to types and numericals can only be `double` or `long` from Netwonsoft - I have a workaround for that in `system-tests`, but asked if it made more sense to do the remapping for more data types like so.

The type we get from OpenTelemetry integrations should be an `int`.

> What happens when we set the status code to 200.0? I'm 80% sure it's Not Good™, let alone if it's OK

Yeah _most_ invalid data types would lead to issues - I could restrict this to checks for natural numbers `short`, `ushort` ... to convert it, but again it should only ever be an `int`

Reply:
Yeah, I mean, I feel like this change is primarily working around our own system-tests, which isn't a great solution IMO. I think the system-tests should have the gnarly decisions and work arounds vs it being in the code here. Especially as I'm pretty sure the backend can't actually handle anything that's not int-like properly (I could have _sworn_ we've seen decimal status codes causing issues)

Reply:
I'll follow up again then with @zarirhamza and @zacharycmontoya for this then, I'd agree that it is a system-test issue more than a tracer issue here.

Reply:
For the most part we only expect the OpenTelemetry AspNet and AspNetCore instrumentations to set this attribute, so let me double-check that and get back to you

Reply:
Latest AspNet [implementation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/c49a8e8faaefca35041e57b6e6faa50ce71c53cc/src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs#L120): Calls `Activity.SetTag("http.response.status_code", (int)statusCode);`

Latest AspNetCore [implementation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L255): Calls `Activity.SetTag("http.response.status_code", (object)((int)statusCode));`

So based on these, we will typically expect that they're set to ints and not as any other numeric type. If that case comes up, we can redirect the application/library author to set it as an int

Reply:
Rewriting the system-tests to match this expected behavior seems fine to be TBH